### PR TITLE
Documentation using mkdocs and adding nightly build file to build site.

### PR DESCRIPTION
**Pull request by aksrajvanshi**
@stu1130 @zachgk @lanking520 @roywei @frankfliu 

Added mkdocs build folder to create documentation. Added nightly build YML file to build website folder.

- Would need additions to the nightly.yml file such as add aws credentials part along with creation of S3 bucket to host the website using Route53.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: nightly build
+
+on:
+  # To trigger this workflow manually, you can use the following curl command:
+  # curl -XPOST -u "USERNAME:PERSONAL_TOKEN" -H "Accept: application/vnd.github.everest-preview+json" -H "Content-Type: application/json" https://api.github.com/repos/awslabs/djl/dispatches --data '{"event_type": "nightly-build"}'
```
Comment (frankfliu) [inline .github/workflows/nightly.yml:5]: https://api.github.com/repos/awslabs/djl-demo/dispatches

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Set up Python3
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.x'
+      - name: install Python Dependencies
+        run: pip3 install mkdocs mkdocs-exclude mknotebooks mkdocs-material jupyter Pygments
+      - name: Install IJava kernel
```
Comment (frankfliu) [inline .github/workflows/nightly.yml:29]: You don't need IJava, since there is no jupyter notebook

File: docs/docs/index.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Welcome to MkDocs
```
Comment (frankfliu) [inline docs/docs/index.md:1]: 1. this file should be README.md
2. This folder should the default docs file for the whole demo project, not just MkDocs

File: docs/mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+google_analytics:
+  - UA-159559852-2
+  - auto
+docs_dir: '../'
+plugins:
+  - search
+  - mknotebooks
+  - exclude:
+      regex:
+        - '.*\.(symbol|params|pt|jar|gz|zip|txt|json|sh|bat|cmd|puml|g4|java|py|cc|h|pro|properties|xml|gradlew|gradle|Dockerfile|RepositoryFactory|EngineProvider|ZooProvider)$|(docker|integration|testing|tools)/.*$'
```
Comment (frankfliu) [inline docs/mkdocs.yml:25]: '.*\.(*+-runner|DS_Store|MF|analysis|ap_|apk|bat|bin|cache|class|classpath|cmd|csv|dat|dat.*|dex|dumpstream|editorconfig|eslintrc|flat|gitignore|gradle|html|idx|iml|index|jar|java|jks|lock|lst|mapping|name|pack|params|pb|prefs|prettierignore|prettierrc|pro|project|properties|pt|sample|sbt|scala|ser|sh|snap|so|ts|tsx|txt|xml)$'

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+nav:
+  - HOME: 'README.md'
+  - Android:
```
Comment (frankfliu) [inline docs/mkdocs.yml:29]: Too many top level menu bar, it may not file in screen

File: docs/mkdocs.yml (hunk 4 above)
Comment (aksrajvanshi) [reply docs/mkdocs.yml:25]: Are these additions to the existing types or should I replace them ?

Comment (frankfliu) [reply docs/mkdocs.yml:25]: those are file extensions for this repo. You need replace with new one:
```
'.*(*+-runner|DS_Store|MF|analysis|ap_|apk|bat|bin|cache|class|classpath|cmd|csv|dat|dat.*|dex|dumpstream|editorconfig|eslintrc|flat|gitignore|gradle|html|idx|iml|index|jar|java|jks|lock|lst|mapping|name|pack|params|pb|prefs|prettierignore|prettierrc|pro|project|properties|pt|sample|sbt|scala|ser|sh|snap|so|ts|tsx|txt|xml)$'
```

Comment (aksrajvanshi) [reply docs/mkdocs.yml:25]: So I think this is invalid. Maybe we don't need the *+ inside the file extensions group. 

Checked it here. regexr.com/597s1

Comment (frankfliu) [reply docs/mkdocs.yml:25]: This should work:
```
.*((.+-runner)|DS_Store|MF|analysis|ap_|apk|bat|bin|cache|class|classpath|cmd|csv|dat|(dat.*)|dex|dumpstream|editorconfig|eslintrc|flat|gitignore|gradle|html|idx|iml|index|jar|java|jks|lock|lst|mapping|name|pack|params|pb|prefs|prettierignore|prettierrc|pro|project|properties|pt|sample|sbt|scala|ser|sh|snap|so|ts|tsx|txt|xml)$
```

File: docs/android/android.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# DJL Android Demo
```
Comment (frankfliu) [inline docs/android/android.md:1]: We should avoid copy README.md file into these folder.

File: .github/workflows/nightly.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    - cron:  '0 10 * * *'
+
+jobs:
+  documentation:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```
Comment (roywei) [inline .github/workflows/nightly.yml:22]: Do you need java here? we don't have any notebook to run here

File: docs/mkdocs.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        - '.*((.+-runner)|DS_Store|MF|analysis|ap_|apk|bat|bin|cache|class|classpath|cmd|csv|dat|(dat.*)|dex|dumpstream|editorconfig|eslintrc|flat|gitignore|gradle|html|idx|iml|index|jar|java|jks|lock|lst|mapping|name|pack|params|pb|prefs|prettierignore|prettierrc|pro|project|properties|pt|sample|sbt|scala|ser|sh|snap|so|ts|tsx|txt|xml)$'
+
+nav:
+  - HOME: 'README.md'
+  - AWS:
+      - AWS-kinesis-video-streams: 'aws/aws-kinesis-video-streams/README.md'
+      - Model Serving on AWS BeanStalk EC2: 'aws/beanstalk-model-serving/README.md'
+      - Serverless Model Serving with DJL: 'aws/lambda-model-serving/README.md'
+  - EcoSystem:
+      - Java Integrations:
```
Comment (roywei) [inline docs/mkdocs.yml:46]: Do we need another level of `java integration`? what else could be there under ecosystem?

File: .github/workflows/nightly.yml (hunk 7 above)
Comment (aksrajvanshi) [reply .github/workflows/nightly.yml:22]: We have different java integrations in there right ? Do we maybe need anything in there right now or probably need in the future ?

File: docs/mkdocs.yml (hunk 8 above)
Comment (aksrajvanshi) [reply docs/mkdocs.yml:46]: We can further divide it into Spring Boot, Camel-DJL, Quarkus, GraalVM if needed.

File: .github/workflows/nightly.yml (hunk 7 above)
Comment (lanking520) [reply .github/workflows/nightly.yml:22]: As long as we don't have notebook to run, we don't need java to build the site

Comment (aksrajvanshi) [reply .github/workflows/nightly.yml:22]: Ok. I'll make changes to it then.

Comment (roywei) [reply .github/workflows/nightly.yml:22]: Let's remove it as unnecessary, all we need is python to convert md files to html. We will probably keep notebook in main repo.

File: docs/mkdocs.yml (hunk 8 above)
Comment (roywei) [reply docs/mkdocs.yml:46]: we can keep it as it for now.

**Pull request merged**